spark/src/main/java/org/apache/iceberg/actions/BaseAction.java

```diff
@@ -30,8 +31,11 @@ protected String metadataTableName(MetadataTableType type) {
     String tableName = table().toString();
     if (tableName.contains("/")) {
       return tableName + "#" + type;
-    } else if (tableName.startsWith("hadoop.") || tableName.startsWith("hive.")) {
-      // HiveCatalog and HadoopCatalog prepend a logical name which we need to drop for Spark 2.4
+    } else if (tableName.startsWith("hadoop.")) {
+      // HadoopCatalog tableName style is 'hadoop.ns.tb'
+      return ((BaseTable) table()).operations().current().location() + "#" + type;
```

Comment:
Why not use `table().location()` instead? There's no need to cast to `BaseTable` and access `TableOperations`.

Reply:
That sounds like a good idea to me. I'll fix it.